### DISCUSSION TITLE
Multiple alternative filters possible

### DISCUSSION THREAD

**Discussion by truthz03** [Mimetis/Dotmim.Sync#1274] [General]
Hi,

I have a scenario where I need a filter to my table (`FileData`) but the UserId is only available inside one of its parent tables.
I know that I can use `filter.AddJoin` to get up to my parent table but what if the are multiple ways to a parent?

![image](https://github.com/user-attachments/assets/abb07a0c-a9b3-44aa-acd0-a940e250c6d5)

In my example I need to check if `ObjectsecurityAttachment` or `FormAttachment` contains the right `FileId` and then use the `UserId` from the `Objectsecurity` or `Form` table.

Is this possible in any way with the latest version of Dotmim.Sync (1.3)?

**Comment by Mimetis** [comment]
using `CustomWhere` ?

**Reply by truthz03** [reply]
Is there an example how to use it?
Do I have to use `AddJoin` together with `CushomWhere`?

**Reply by Mimetis** [reply]
No there is no `AddCustomJoin`

Usually, in that case, I think that `AddCustomWhere` should contains a Where clause that have a subquery in it
like

`And UserId in (Select ...... JOIN .... JOIN ... Where ....)`

**Reply by truthz03** [reply]
I get it working.

If anybody else is interessted I do the following things:
1) Use `filter.AddJoin` like before for all parent tables
2) Use `filter.AddParameter` only once per column name (in my case `UserId` is used in all top level tables)
3) Use `filter.AddCustomWhere` only one with the following parameter
```
"(
    [base].[InstanceId] = 'xyz' 
    AND 
    (
        [Objectsecurity].[UserId] = @UserId 
        OR 
        [Form].[UserId] = @UserId
    )
) 
OR 
[side].[sync_row_is_tombstone] = 1"
```

In my example:
`[side]` is the `FileData_tracking` table
`[base]` is the `FileData` table
joined tables can be used with its real name like `[Objectsecurity]` or `[Form]`

**Reply by Mimetis** [reply]
Don't forget to handle deleted rows : https://dotmimsync.readthedocs.io/Filters.html#how-to-get-deleted-rows-in-your-where-clause

**Reply by truthz03** [reply]
Ok. Then I miss something. 
This seems to be the problem [here](https://github.com/Mimetis/Dotmim.Sync/issues/1275)

**Reply by truthz03** [reply]
I updated my answer above to give a working solution :)
